src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -420,7 +414,13 @@ void SolverInterfaceImpl::advance(
   // Update the coupling scheme time state. Necessary to get correct remainder.
   _couplingScheme->addComputedTime(computedTimeStepSize);
   // Current time
-  double time = _couplingScheme->getTime();
+  const double time          = _couplingScheme->getTime();
+  const double relativeTime  = _couplingScheme->getNormalizedWindowTime();
+  const bool   isAtWindowEnd = math::equals(relativeTime, time::Storage::WINDOW_END);
```

Comment:
Sounds reasonable. This is a good reason to put the whole thing inside `CouplingScheme`, since otherwise we would need to expose `_eps`. See 6b39b1ad441141e70cc0c7201b98daf694759b0f.